lib/log_decorator/log_proxy.rb

```diff
@@ -0,0 +1,50 @@
+module LogDecorator
+  class LogProxy
+    attr_accessor :prefix, :logger
+
+    def self.initialize_log_levels # rubocop:disable AbcSize
+      return if @log_levels_initialized
+      @log_levels_initialized = true
+
+      LogDecorator.log_levels.each do |level|
+        define_method(level) do |msg = nil, &blk|
```

Comment:
This might be better handled by mixing in a module with the defined methods as opposed to defining them on the target, but that can also be a PR.  (This is the pattern that ActiveRecord uses for column and relation methods, and that we've reused elsewhere for things like virtual columns).